src/framework/components/anim/component-layer.js

```diff
@@ -10,12 +11,16 @@ import { AnimTransition } from '../../../anim/controller/anim-transition.js';
  * @param {string} name - The name of the layer.
  * @param {object} controller - The controller to manage this layers animations.
  * @param {AnimComponent} component - The component that this layer is a member of.
+ * @param {number} weight - The weight of this layer. Defaults to 1.
  */
 class AnimComponentLayer {
-    constructor(name, controller, component) {
+    constructor(name, controller, component, index, weight = 1, blendType = ANIM_LAYER_OVERWRITE) {
```

Comment:
It's ok to take blendType as constructor parameter, but we need a setter for it for the case the user needs to change it later.

Reply:
Changing the blendType at runtime would require updating the layer mask contained in each AnimTargetValue as its based on the unique combination of blendTypes for all layers associated with that AnimTargetValue. I'm not sure this is worth adding in at this point.

Reply:
fair enough